### PR TITLE
Update Currency.php

### PR DESCRIPTION
**Pull request by ghaliano**
adding __toString method

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Catchable error on form combination generation ("Catchable Fatal Error: Object of class Currency could not be converted to string").
| Type?         | improvement
| Category?     | BO 
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Please indicate how to best verify that this PR is correct.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16372)

### CODE REVIEW

**Comment by Progi1984** [comment]
@ghaliano Thank you for your contribution. Could you give us a scenario test for testing this PR ? Could you open an issue for the issue related to this PR, please ? Thanks

File: classes/Currency.php [hunk 1]
```diff
@@ -866,4 +866,15 @@ public function refreshLocalizedCurrencyData(array $languages, LocaleRepository
         $this->name = $namesByLang;
         $this->symbol = $symbolsByLang;
     }
+    
+
+    /**
+     * Classic __toString php behaviour
+     * @return string
```
Comment (Progi1984) [inline classes/Currency.php:873]:
```suggestion
     *
     * @return string
```

File: classes/Currency.php [hunk 2]
```diff
@@ -866,4 +866,15 @@ public function refreshLocalizedCurrencyData(array $languages, LocaleRepository
...
+     */
+    public function __toString()
+    {
+      return (string) $this->name;
```
Comment (Progi1984) [inline classes/Currency.php:877]:
```suggestion
        return (string) $this->name;
```

File: classes/Currency.php [hunk 3]
```diff
@@ -866,4 +866,15 @@ public function refreshLocalizedCurrencyData(array $languages, LocaleRepository
...
+    }
+
```
Comment (Progi1984) [inline classes/Currency.php:879]:
```suggestion
```

**Comment by ghaliano** [comment]
@Progi1984  sorry for the delay i can't find the time to do a complete reproduction of this bug
otherwise it's very simple to reproduce it, just try to add a new combination, the combination form  did not show up.
best regard

**Comment by PierreRambaud** [comment]
Hi @ghaliano 
Feel free to reopen this pull request when you will have time to work on ti :wink: 

Kind regards

**Pull request closed without merging**